scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -177,7 +198,10 @@ final case class SpecificRecordIO[T <: SpecificRecord: ClassTag: Coder](path: St
         params.suffix,
         params.codec,
         params.metadata,
-        params.tempDirectory
+        params.shardNameTemplate,
+        ScioUtil.tempDirOrDefault(params.tempDirectory, data.context),
```

Comment:
Here is a choice I am not 100% on; do we want to use the pipeline default temp dir for the write when none is provided (as we do in SMB) or is there a better choice?